ui/src/components/app-form/app-form.tsx

```diff
@@ -486,45 +539,154 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
         <Controller
           name="framework"
           control={control}
-          rules={REQUIRED_FORM_FIELDS_RULES}
+          rules={{ required: true }}
           // eslint-disable-next-line @typescript-eslint/no-unused-vars
-          render={({ field: { ref: _, ...field } }) => (
-            <FormControl>
-              <InputLabel id="framework-label" required>
-                Framework
+          render={({ field: { ref, ...field } }) => (
+            <FormControl error={!!errors.framework}>
+              {errors.framework && (
+                <Box
+                  display="flex"
+                  alignItems="center"
+                  color="error.main"
+                  mb={2}
+                >
+                  <ErrorIcon fontSize="small" />
+                  <Typography variant="body2" color="error" ml={1}>
+                    Select a framework
+                  </Typography>
+                </Box>
+              )}
+              <InputLabel
+                id="framework-label"
+                sx={{
+                  fontSize: '1rem',
+                  ...(errors.framework && {
+                    transform: 'translate(14px, 27px) scale(0.75)',
+                    backgroundColor: '#fafafa',
+                    padding: '0 .25rem',
+                  }),
+                }}
+              >
+                <CustomLabel label="Framework" required={true} />
               </InputLabel>
               <Select
                 {...field}
                 id="framework"
-                label="Framework"
-                required
-                error={errors.framework?.message ? true : false}
+                error={!!errors.framework}
+                displayEmpty
+                inputProps={{ 'aria-label': 'Select framework' }}
               >
+                {errors.framework && (
+                  <MenuItem
+                    value=""
+                    disabled
+                    sx={{ color: 'limegreen' }}
```

Comment:
Is this supposed to be lime green?

Reply:
No. Was looking for an obvious change in color for testing.